src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -114,11 +114,31 @@ public static ParameterizedTypeName get(ClassName rawType, TypeName... typeArgum
     return new ParameterizedTypeName(null, rawType, Arrays.asList(typeArguments));
   }
 
+  /** Returns a parameterized type, applying {@code typeArguments} to {@code rawType}. */
+  public static ParameterizedTypeName get(
+      ClassName rawType, Iterable<? extends TypeName> typeArguments) {
+    ArrayList<TypeName> typeArgumentsList = new ArrayList<>();
+    for (TypeName typeArgument : typeArguments) {
+      typeArgumentsList.add(typeArgument);
+    }
+    return get(rawType, typeArgumentsList.toArray(new TypeName[0]));
```

Comment:
Why are you calling toArray instead of calling the constructor directly?

Reply:
Ah this is copy/pasta. This overload used to do that, but I couldn't do that with the other one since it needs to pass through the `list(..)` helper which only accepts an array and changing that to accept a `List` is far more invasive.

Reply:
Fixed